### ISSUE TITLE
transformer: incorrect `isStaticChildren` value for fragments with static children

### ISSUE THREAD

**Issue by sapphi-red** [oxc-project/oxc#8650]
For input,
```jsx
const foo = () => {
  return <>
    <p>foo</p>
    <p>foo</p>
  </>
}
```
oxc tranform outputs:
```js
var _jsxFileName = "foo.jsx";
import { jsxDEV as _jsxDEV, Fragment as _Fragment } from "react/jsx-dev-runtime";
const foo = () => {
        return _jsxDEV(_Fragment, { children: [_jsxDEV("p", { children: "foo" }, void 0, false, {
                fileName: _jsxFileName,
                lineNumber: 4,
                columnNumber: 5
        }, this), _jsxDEV("p", { children: "foo" }, void 0, false, {
                fileName: _jsxFileName,
                lineNumber: 5,
                columnNumber: 5
        }, this)] }, void 0, false);
};
```
The forth argument of `_jsxDEV` for `_Fragment` is **`false`**. This should be true.
[reproduction using `transform` function from rolldown](https://stackblitz.com/edit/github-xv4yss-marnqaiv?file=build.mjs)

If this is `false`, `Warning: Each child in a list should have a unique "key" prop.` error happens with React.

[esbuild try with the same input](https://esbuild.github.io/try/#dAAwLjI0LjIALS1sb2FkZXI9anN4IC0tanN4PWF1dG9tYXRpYyAtLWpzeC1kZXYAY29uc3QgZm9vID0gKCkgPT4gewogIHJldHVybiA8PgogICAgPHA+Zm9vPC9wPgogICAgPHA+Zm9vPC9wPgogIDwvPgp9)

I found this while testing rolldown-vite on ecosystem-ci.

**Comment by sapphi-red** [comment]
fixed by #8713
